<a href="https://colab.research.google.com/github/sr606/Python-Practice/blob/main/Aerofit_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Defining Problem Statement and Analysing basic metrics**


###**About Aerofit**

Aerofit is a leading brand in the field of fitness quipment. Aerofit provides a product range including machines such as treadmills, exercise bikes, gym equipment, and fitness accessories to cater the needs of all categories of people.

###**Business Problem**

The market research team at AeroFit wants identify the characteristics of the target audience for each type of treadmill offered by the company, to provide a better recommendation of the treadmills to new customers. The team decides to investigate whether there are differences across the product with respect to customer characteristics.

1. Perform descriptive analytics to create a customer profile for each AeroFit treadmill product by developing appropriate tables and charts.

2. For each AeroFit treadmill product, construct two-way contingency tables and compute all conditional and marginal probabilities along with their insights/impact on the business.

##**Imposrting Libraries**

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt # data visualisation library
import seaborn as sns #data visualisation librabry based on matplotlib
from scipy import stats
from scipy.stats import norm


##**Importing the dataset**

In [2]:
! gdown 1jyIrDhgS-OKwby2mctMIGOFyrXG2VMec

Downloading...
From: https://drive.google.com/uc?id=1jyIrDhgS-OKwby2mctMIGOFyrXG2VMec
To: /content/aerofit_treadmill.csv
100% 7.28k/7.28k [00:00<00:00, 23.0MB/s]


In [3]:
df=pd.read_csv('/content/aerofit_treadmill.csv')
df

Product  Age  Gender  Education MaritalStatus  Usage  Fitness  Income  \
0     KP281   18    Male         14        Single      3        4   29562   
1     KP281   19    Male         15        Single      2        3   31836   
2     KP281   19  Female         14     Partnered      4        3   30699   
3     KP281   19    Male         12        Single      3        3   32973   
4     KP281   20    Male         13     Partnered      4        2   35247   
..      ...  ...     ...        ...           ...    ...      ...     ...   
175   KP781   40    Male         21        Single      6        5   83416   
176   KP781   42    Male         18        Single      5        4   89641   
177   KP781   45    Male         16        Single      5        5   90886   
178   KP781   47    Male         18     Partnered      4        5  104581   
179   KP781   48    Male         18     Partnered      4        5   95508   

     Miles  
0      112  
1       75  
2       66  
3       85  
4       47  
..     ...  
175    200  
176    200  
177    160  
178    120  
179    180  

[180 rows x 9 columns]

##**Baisc Analysis**

###**Shape of the data**

In [4]:
df.shape

(180, 9)

###**Columns in DataFrame**

In [5]:
df.columns

Index(['Product', 'Age', 'Gender', 'Education', 'MaritalStatus', 'Usage',
       'Fitness', 'Income', 'Miles'],
      dtype='object')

####**First 5 rows**

In [9]:
df.head(5)

Product  Age  Gender  Education MaritalStatus  Usage  Fitness  Income  Miles
0   KP281   18    Male         14        Single      3        4   29562    112
1   KP281   19    Male         15        Single      2        3   31836     75
2   KP281   19  Female         14     Partnered      4        3   30699     66
3   KP281   19    Male         12        Single      3        3   32973     85
4   KP281   20    Male         13     Partnered      4        2   35247     47

###**Data Type of all attributes(columns)**

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Product        180 non-null    object
 1   Age            180 non-null    int64 
 2   Gender         180 non-null    object
 3   Education      180 non-null    int64 
 4   MaritalStatus  180 non-null    object
 5   Usage          180 non-null    int64 
 6   Fitness        180 non-null    int64 
 7   Income         180 non-null    int64 
 8   Miles          180 non-null    int64 
dtypes: int64(6), object(3)
memory usage: 12.8+ KB


####**Analysis**



1.   There is no missing values in dataframe
2.   Most of the columns are numeric except Product, Gender and Marital Status



###**Statistical Summary**
####**Statastical Summary of Numeric Columns**

In [11]:
df.describe().T

count          mean           std      min       25%      50%  \
Age        180.0     28.788889      6.943498     18.0     24.00     26.0   
Education  180.0     15.572222      1.617055     12.0     14.00     16.0   
Usage      180.0      3.455556      1.084797      2.0      3.00      3.0   
Fitness    180.0      3.311111      0.958869      1.0      3.00      3.0   
Income     180.0  53719.577778  16506.684226  29562.0  44058.75  50596.5   
Miles      180.0    103.194444     51.863605     21.0     66.00     94.0   

                75%       max  
Age           33.00      50.0  
Education     16.00      21.0  
Usage          4.00       7.0  
Fitness        4.00       5.0  
Income     58668.00  104581.0  
Miles        114.75     360.0

####**Statistical Summary of Object Columns**

In [14]:
df.describe(include = 'object').T

count unique        top freq
Product         180      3      KP281   80
Gender          180      2       Male  104
MaritalStatus   180      2  Partnered  107

Here we can see most of the users are using **KP281** machine and its frequency or count is 80 out of 180 total counts

And the product is mostly has most of the users from male category


##**Non-Graphical Analysis: Value counts and unique attributes**

###**Product Column**


####**Unique**

In [15]:
df['Product'].unique()

array(['KP281', 'KP481', 'KP781'], dtype=object)

Aerofit produces three treadmill models: KP281,KP481, and KP781

####**value counts**

In [17]:
Product_counts = df['Product'].value_counts(normalize = True)*100
Product_counts

Product
KP281    44.444444
KP481    33.333333
KP781    22.222222
Name: proportion, dtype: float64

**Insight:**

Among the users, 44.44% prefer using KP281 Treadmill, while 33.33% users prefer KP481 treadmill, and only 22.22% people favor the KP871 treadmill.